src/lib/random-weighted-choices.ts

```diff
@@ -0,0 +1,98 @@
+/**
```

Comment:
This algorithm seems good, but I'd suggest putting it through o1 and asking it to add detailed comments explaining the logic. We should also include a link to the original library and preserve its license here.

Reply:
I'll add a link to the package license field, it's the only documented license of the library, there's no license file or preamble present in the repo. Do we want to copy paste MIT license into the file, or just leave a link to the license field?
https://github.com/parmentf/random-weighted-choice/blob/626398ce68df3269996e05cc0d7e525f4a83322a/package.json#L40

Reply:
I had o1 add comments

Reply:
done